libs/core/langchain_core/messages/__init__.py

```diff
@@ -98,6 +98,15 @@ def messages_from_dict(messages: Sequence[dict]) -> List[BaseMessage]:
     return [_message_from_dict(m) for m in messages]
 
 
+def message_chunk_to_message(chunk: BaseMessageChunk) -> BaseMessage:
+    if not isinstance(chunk, BaseMessageChunk):
+        return chunk
+    # chunk classes always have the equivalent non-chunk class as their first parent
+    return chunk.__class__.__mro__[1](
```

Comment:
feels a bit magical -- any reason not to rely on instance checks to make the code a bit less scary?